Extensions/CodeGeneration/CodeGenerationDisplayDriver.cs

```diff
@@ -133,16 +133,18 @@ private string ConvertJToken(JToken jToken)
                     };
 
                 case JArray jArray:
-                    var token = string.Join(", ", jArray.Select(ConvertJToken));
-                    var format = token.Contains("ListValueOption") ? "\n    " : string.Empty;
-
+                    var token = string.Join(", ", jArray.Select(a => ConvertJToken(a, indentationDepth * 2)));
+                    var format = token.Contains("ListValueOption", StringComparison.InvariantCulture)
+                        ? $"\n{string.Join(string.Empty, Enumerable.Repeat(" ", (int)(indentationDepth * 1.5)))}"
+                        : string.Empty;
 
                     return $"new[] {format}{{ {string.Join(", ", token)} {format}}}";
 
                 case JObject jObject:
+                    var indentation = string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth));
                     if (jObject["name"] != null && jObject["value"] != null)
                     {
-                        return $"\n               new ListValueOption {{ Name = \"{jObject["name"]}\", Value = \"{jObject["value"]}\" }}";
+                        return $"\n{indentation}new ListValueOption {{ Name = \"{jObject["name"]}\", Value = \"{jObject["value"]}\" }}";

```

Comment:
Make this rather output a linebreak-structured class, like the others. I.e. instead of:

```c#
new ListValueOption { Name = "Option1", Value = "Option1" }
``

It should output (also note the trailing comma):

```c#
new ListValueOption
{
    Name = "Option1",
    Value = "Option1",
}
```

Reply:
Done.

![image](https://user-images.githubusercontent.com/237369/157143061-9ae6e8ca-004d-477b-9eed-1982f91f6218.png)

Reply:
There's an excess bracket after each `Value`.

Reply:
Good catch. Fixed.

![image](https://user-images.githubusercontent.com/237369/157249325-af4de817-a17e-4dd6-aedb-71462272572f.png)

Reply:
The `@` in front of `new[]` seems out of place. Please copy the code into a C# file to see if it's syntactically correct.

Reply:
Ok I had to add a little logic to exclude the `@` sign. Let me know what you think.

```
_contentDefinitionManager.AlterPartDefinition("Test", part => part
    .WithField("Predefined", field => field
        .OfType("TextField")
        .WithDisplayName("Predefined")
        .WithEditor("PredefinedList")
        .WithPosition("0")
        .WithSettings(new TextFieldPredefinedListEditorSettings
        {
            Options = new[]
            {
                new ListValueOption
                {
                    Name = "One",
                    Value = "1",
                },
                new ListValueOption
                {
                    Name = "Two",
                    Value = "2",
                },
                new ListValueOption
                {
                    Name = "Three",
                    Value = "3",
                },
            },
            DefaultValue = "1",
        })
    )
);

_contentDefinitionManager.AlterPartDefinition("ContentMenuItemPart", part => part
    .Attachable()
        .WithDisplayName("Content Menu Item")
        .WithDescription("Provides a menu link behavior with the Content Picker for your content item.")
    .WithField("SelectedContentItem", field => field
        .OfType("ContentPickerField")
        .WithDisplayName("Selected Content Item")
        .WithPosition("0")
        .WithSettings(new ContentPickerFieldSettings
        {
            Required = true,
            DisplayAllContentTypes = true,
            DisplayedContentTypes = new[]
            {
            },
        })
    )
);
```